packages/rum/test/specs/apm-base.spec.js

```diff
@@ -83,6 +107,43 @@ describe('ApmBase', function () {
     })
   })
 
+  it('should send page load metrics if page is backgrounded before load event is triggered', done => {
+    // Make sure page load event logic is not executed when initializing the agent.
+    // It's important to realize that when this test is executed Karma setup is already loaded in the browser.
+    // Hence, the page load event would be already executed.
+    // Faking the document.readyState makes possible to test the expected scenario
+    function setDocumentReadyState(state) {
+      Object.defineProperty(document, 'readyState', {
+        get() {
+          return state
+        }
+      })
+    }
+    const originalState = document.readyState
+    setDocumentReadyState('loading')
+
+    apmBase.init({ serviceName, serverUrl })
+
+    var tr = apmBase.getCurrentTransaction()
+
+    expect(tr.name).toBe('Unknown')
+    expect(tr.type).toBe(PAGE_LOAD)
+
+    hidePageSynthetically('visibilitychange')
```

Comment:
Despite `visibilitychange` being the default value of `hidePgeSynthetically` I added it explicitly because my intent is to help developers reading this to understand that is entwined with the beforeEach logic